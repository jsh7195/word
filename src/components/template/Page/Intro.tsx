import React from 'react';
import useFetch from "@hooks/useFetch";
import Button from "@atoms/Button";

const Intro = () => {
  const [data, loading] = useFetch("/get-words");
  sessionStorage.setItem("words", JSON.stringify(data));

  return (
    <div>
      <div>
        <h2>문제 단어 나오는곳</h2>
      </div>
      <div>{loading ? "Loading..." : <Button text="시작" url="/Play" />}</div>
    </div>
  );
};

export default Intro;
