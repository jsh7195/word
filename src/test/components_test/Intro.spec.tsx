import React from "react";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import Intro from "@template/Page/Intro";

Enzyme.configure({ adapter: new Adapter() });

test("Intro Render Test ... <Intro/>", () => {
  const component = renderer
    .create(
      <BrowserRouter>
        <Intro />
      </BrowserRouter>
    )
    .toJSON();
  expect(component).toMatchSnapshot();
});
