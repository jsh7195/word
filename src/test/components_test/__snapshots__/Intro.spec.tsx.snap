// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intro Render Test ... <Intro/> 1`] = `
<div>
  <div>
    <h5>
      문제 단어
    </h5>
  </div>
  <div>
    <button>
      <a
        href="/Play"
        onClick={[Function]}
      >
        시작
      </a>
    </button>
  </div>
</div>
`;
